### DISCUSSION TITLE
Code generation on locally run service

### DISCUSSION THREAD

**Discussion by DrSensor** [encoredev/encore#247] [Help] [answered]
Is there a way to make code generation works without specifying `--env=`?
I want to generate the client code in live development when running `encore run`.

**Comment by DrSensor** [comment, marked as answer]
Never mind, just found out I can do `encore gen client $APP_ID --env=local --output ../src/service.ts --lang typescript`

**Reply by magiconair** [reply]
I'm struggling with this. What is the APP_ID? Is this the `local_id` in the manifest, e.g. `a9m3g`? I've created a hello-world app `config-api` is running which is just the HelloWorld example and I've tried `config-api`, `config-api-a9m3g` and `a9m3g` but always get the same error.

```shell
> encore gen client a9m3g --env=local -l ts -o client.ts
error: the app a9m3g must be run locally before generating a client for the 'local' environment.
```
